README.md

```diff
@@ -1,87 +1,86 @@
-# biconomy-paymasters
+# Biconomy Paymasters 🌐
 
-### Playground for ERC4337 Paymasters built with :heart_eyes: by Biconomy 
-#### Some examples 
+Welcome to the `biconomy-paymasters` repository! Paymaster contracts enable seamless transaction fee handling in decentralized applications. 🛠️
 
-1. Singleton Verifying Paymaster: Acts as a sponsorship paymaster and lets Dapps manage deposit without deploying a new one for each Dapp.
+## What is Biconomy Paymasters? 🤔
 
-2. Token Paymaster: Extended version of Verifying Paymaster which can accept fees from user by withdrawing ERC20 tokens
+Biconomy Paymasters are smart contracts that abstract away the complexity of gas fees for end-users. By utilizing these contracts, developers can offer their users gasless transactions or the ability to pay for gas in ERC20 tokens. 🚀
 
-#### Other WIP
+- **Sponsorship Paymaster**: Allows transactions without end-users needing to pay for gas, enhancing UX.
+- **Token Paymaster**: Provides the ability to pay for transactions with ERC20 tokens.
 
-Fiat Paymaster
+## Features 🌟
 
-Deposit Paymaster
+- Simplified transaction fee handling.
+- ERC 4337 Account Abstraction compliant.
+- Multi-token support for gas payments.
 
-Custom versions of above Verifying (allow Dapp deposit sponsorship in different tokens) and Token Paymaster (acts as deposit paymaster also)
+### ERC20 Token Paymaster
 
+![ERC20 Token Paymaster](./assets/readme/erc20-token-gas-payment-flow.png)
 
+- ERC20 Token Paymaster helps users pay for their transactions using ERC20 tokens.
+- Users initiate a transaction using an ERC20 token.
+- Paymaster validates the transaction and forwards it to the network while handling necessary fee conversions.
+- This flow ensures ease of use and convenience for users.
 
-# How to run the project
+### Sponsorship Paymaster
 
-#### You're going to need to place a private key in a .env file in the root. ####
+![Sponsorship Paymaster](./assets/readme/paymaster-off-chain-request-processing-flow.png)
 
-#### In order to add/udpate git submodule account-abstraction: ####
-.gitmodules file is already added. two submodules are being used in this project
+- Sponsorship Paymaster covers transaction fees for users.
```

Comment:
I suggest changing txns to userOps everywhere

Reply:
Maybe I'm missing something, but I find it a bit confusing, that we have it named 'Sponsorship Paymaster', however I can't find Sponsorship Paymaster in the repo.
I understand Verifying Paymaster is not the best name, Sponsorship Paymaster is much better , but I think need some consistency between readme and file names here

Reply:
You're right @filmakarov, I found the references to 'Sponsorship Paymaster' in both Biconomy [blog](https://www.biconomy.io/post/how-biconomy-paymaster-works-sponsor-gas-or-pay-in-any-token) and [official documentation](https://docs.biconomy.io/Paymaster/description#sponsorship-paymaster). 

I agree 'Sponsorship Paymaster' sounds like a better name to me too. Let's bring this up with the team for further discussion.